src/Microsoft.AspNetCore.Razor.Language/Legacy/ParserContext.cs

```diff
@@ -9,14 +9,18 @@ namespace Microsoft.AspNetCore.Razor.Language.Legacy
 {
     internal partial class ParserContext
     {
-        public ParserContext(ITextDocument source, RazorParserOptions options)
+        public ParserContext(RazorSourceDocument source, RazorParserOptions options)
         {
             if (source == null)
             {
                 throw new ArgumentNullException(nameof(source));
             }
 
-            Source = source;
+            SourceDocument = source;
+            var chars = new char[source.Length];
+            source.CopyTo(0, chars, 0, source.Length);
+
+            Source = new SeekableTextReader(chars, source.FileName);
```

Comment:
Before I published this PR I removed the TextReader base type from SeekableTextReader because it technically adds nothing today. Reverted those changes because I didn't want to pollute this PR, but you're right, if it's a TextReader this should be disposed.

Thoughts on me nuking the TextReader basetype?